realm/realm-library/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -120,15 +119,13 @@ public DynamicRealmObject createObject(String className, Object primaryKeyValue)
     /**
      * Adds a change listener to the Realm.
      * <p>
-     * The listeners will be executed:
-     * <ul>
-     * <li>Immediately if a change was committed by the local thread</li>
-     * <li>On every loop of a Handler thread if changes were committed by another thread</li>
-     * <li>On every call to {@link io.realm.Realm#refresh()}</li>
-     * </ul>
+     * The listeners will be executed on every loop of a Handler thread if
+     * changes why committed by this or another thread.
      *
-     * Listeners are stored as a strong reference, you need to remove the added listeners using {@link #removeChangeListener(RealmChangeListener)}
-     * or {@link #removeAllChangeListeners()} which removes all listeners including the ones added via anonymous classes.
+     * Realm instances are cached, so for that reason it is important to
```

Comment:
maybe `are cached per thread`. And it might worth to say that all listeners will be removed from the instance in this thread in the `removeAllChangeListernerers`?